cmake/CPM.cmake

```diff
@@ -712,6 +712,12 @@ function(CPMAddPackage)
     # relative paths.
     get_filename_component(download_directory ${download_directory} ABSOLUTE)
     list(APPEND CPM_ARGS_UNPARSED_ARGUMENTS SOURCE_DIR ${download_directory})
+
+    if(NOT ${CPM_ARGS_NAME}_CPM_HAS_CACHE_LOCK)
```

Comment:
Isn't this (local) variable useless since you are always going to lock and release the file anyways ?
Besides this doesnt need to be done when clones are operated into different directories (i.e. when `CPM_SOURCE_CACHE` is not used, which could be a somewhat decent condition to check ?).

Reply:
Yes. It looks like these variables are useless. The reason it exists is that in the past I had come across an error that occurred because the point of execution entered the locked block then left it and went back into it without going past the unlocking line. This would cause an error along the lines of `file already locked!` to occur. The local variable was used to solve that problem. I just tried it again without the `${CPM_ARGS_NAME}_CPM_HAS_CACHE_LOCK` and it succeeded, so I will happily remove the checks.

As for the `CPM_SOURCE_CACHE` check, a while ago I moved all of my changes to within a block that checked for this. see [line 700](https://github.com/PercentBoat4164/CPM.cmake/blob/a5a33b8141b8e8954f040c13bb22bd49f7aa25ea/cmake/CPM.cmake#L700). If this is not a good place for it, I can move it elsewhere and perform my own checks.